### DISCUSSION TITLE
pylsp not installed

### DISCUSSION THREAD

**Discussion by dave-kennedy** [mason-org/mason.nvim#387] [Q&A]
I must be missing something here:

```
require('packer').startup(function (use)
    use 'wbthomason/packer.nvim'

    use {
        'williamboman/mason-lspconfig.nvim',
        config = function()
            require('mason').setup()

            require('mason-lspconfig').setup({
                automatic_installation = true,
                ensure_installed = {'pylsp'}
            })

            require('lspconfig').pylsp.setup({
                on_attach = on_attach
            })
        end,
        requires = {
            'williamboman/mason.nvim',
            'neovim/nvim-lspconfig'
        }
    }
```

I'm new to Packer and Mason. Pretty sure Packer is working fine, but I must be missing something in the Mason config because I get this error when opening a Python file: `Spawning language server with cmd: `pylsp` failed. The language server is either not installed, missing from PATH, or not executable.`

Confirmed there is no such executable in the .local/share/nvim/mason directory.

Edit: I deleted the .local/share/nvim/mason directory, recreated packer_compiled.lua and restarted nvim. It took a few minutes, but eventually an error popped up:

```
[WARN  Tue 06 Sep 2022 08:50:53 PM MDT] .../packer/start/mason.nvim/lua/mason-core/EventEmitter.lua:23: EventEmitter handler failed for event "closed" with error "...ck/packer/start/nvim-lspconfig/lua/lspconfig/configs.lua:34: E5560: nvim_create_augroup must not be called in a lua loop callback"
```

I restarted again and now it works.

**Comment by williamboman** [comment]
Hello! Thanks for sharing the log entry, I believe that's an error with Mason. I've opened https://github.com/williamboman/mason-lspconfig.nvim/pull/50 which hopefully fixes this.
